src/main/java/com/project/poorlex/api/service/BattleService.java

```diff
@@ -74,4 +84,85 @@ public BattleSearchResponse searchBattle(BattleFilter filter) {
+
+    // 배틀 방 상세 조회
+    public BattleDetailResponse searchBattleDetail(Long battleRoomId){
+
+        Optional<Battle> battleOptional = battleRepository.findById(battleRoomId);
+
+        List<BattleUser> battleUsers = battleUserRepository.findByBattleId(battleRoomId);
+
+        if(battleOptional.isEmpty()){
+            throw new BattleCustomException(BattleErrorCode.BATTLE_NOT_FOUND);
+        }
+
+        Battle battle = battleOptional.get();
+
+        BattleDetailResponse response = new BattleDetailResponse();
+        response.setId(battle.getId());
+        response.setName(battle.getName());
+        response.setBudget(battle.getBudget());
+        response.setTotal(battle.getTotal());
+
+        List<Long> memberIds = battleUsers.stream()
+                .map(battleUser -> battleUser.getMember().getId())
+                .collect(Collectors.toList());
+
+        List<Member> members = memberRepository.findAllById(memberIds);
+
+        List<BattleMemberResponse> battleMembers = members.stream()
+                .map(member -> {
+                    BattleMemberResponse battleMemberResponse = new BattleMemberResponse();
+                    battleMemberResponse.setId(member.getId());
+                    battleMemberResponse.setName(member.getName());
+                    return battleMemberResponse;
+                }).collect(Collectors.toList());
+
+        // 진행 전인 방인 경우
+        if(battle.getBattleStatus() == BattleStatus.WAITING){
+            response.setDescription(battle.getDescription());
+            response.setBattleMembers(battleMembers);
+        }
+        else if(battle.getBattleStatus() == BattleStatus.PROGRESS){
+            // TODO : 지출 작업 후 진행
+        }
+
+        return response;
+    }
+
+    // 배틀 참여하기
+    @Transactional
+    public BattleJoinResponse joinBattle(Long battleRoomId){
+
+        Member member = authService.findMemberFromToken();
+
+        Optional<Battle> battleOptional = battleRepository.findById(battleRoomId);
+
+        if(battleOptional.isPresent()){
+
+            Battle battle = battleOptional.get();
+
+            if(battle.getBattleStatus() == BattleStatus.PROGRESS){
+                throw new BattleCustomException(BattleErrorCode.BATTLE_IS_ALREADY_IN_PROGRESS);
+            }
+            else if(battle.getBattleStatus() == BattleStatus.FINISHED){
+                throw new BattleCustomException(BattleErrorCode.BATTLE_IS_ALREADY_FINISH);
+            }
+
+            BattleUser battleUser = BattleUser.builder()
+                    .member(member)
+                    .battle(battleOptional.get())
+                    .role(Role.USER)
+                    .build();
+
+            battleUserRepository.save(battleUser);
+
+            BattleJoinResponse response = new BattleJoinResponse();
+            response.setBattleId(battle.getId());
+            return response;
+        }
+        else{
+            throw new BattleCustomException(BattleErrorCode.BATTLE_NOT_FOUND);
+        }
```

Comment:
1. `optionalBattle.get()`와 같은 if문으로 분기하기보단 아래처럼 나타내는게 더 직관적입니다!
 2. Response도 Builder를 통해 나타내어 `BattleDetailResponse`와 같이 컨벤션 맞춰주시는게 좋아보이네요
```suggestion
        Battle battle= battleRepository.findById(battleRoomId)
            .orElseThrow(() -> new BattleCustomException(BattleErrorCode.BATTLE_NOT_FOUND));
  
            if(battle.getBattleStatus() == BattleStatus.PROGRESS){
                throw new BattleCustomException(BattleErrorCode.BATTLE_IS_ALREADY_IN_PROGRESS);
            } else if(battle.getBattleStatus() == BattleStatus.FINISHED){
                throw new BattleCustomException(BattleErrorCode.BATTLE_IS_ALREADY_FINISH);
            }

            BattleUser battleUser = BattleUser.builder()
                    .member(member)
                    .battle(battle)
                    .role(Role.USER)
                    .build();

            battleUserRepository.save(battleUser);

            BattleJoinResponse response = new BattleJoinResponse();
            response.setBattleId(battle.getId());
            return response;
```

Reply:
변경하였습니다 감사합니다!!